Xamarin.PropertyEditing/ViewModels/PropertyViewModel.cs

```diff
@@ -359,6 +407,49 @@ private async void OnCreateBinding ()
 			});
 		}
 
+		private async void UpdateAutocomplete (string value)
+		{
+			if (!SupportsAutocomplete)
+				return;
+
+			if (this.autocomplete == null) {
+				this.autocomplete = new ObservableCollectionEx<string> ();
+				OnPropertyChanged (nameof(AutocompleteItems));
+			} else {
+				this.autocompleteCancel.Cancel();
+			}
+
+			this.autocompleteCancel = new CancellationTokenSource ();
+			CancellationToken cancel = this.autocompleteCancel.Token;
+
+			try {
+				HashSet<string> common = null;
+				List<Task<IReadOnlyList<string>>> tasks = new List<Task<IReadOnlyList<string>>> ();
+
+				foreach (IObjectEditor editor in Editors) {
+					if (!(editor is ICompleteValues complete))
+						continue;
+
+					tasks.Add (complete.GetCompletionsAsync (Property, value, cancel));
+				}
+
+				IReadOnlyList<string> list = null;
+				do {
+					Task<IReadOnlyList<string>> results = await Task.WhenAny (tasks);
+					tasks.Remove (results);
+
+					if (list == null) {
+						list = results.Result;
+						common = new HashSet<string> (list);
+					} else
+						common.IntersectWith (results.Result);
+				} while (tasks.Count > 0 && !cancel.IsCancellationRequested);
+
+				this.autocomplete.Reset (list.Where (common.Contains));
```

Comment:
It looks like when this is called, the autocomplete will be the intersection of the first list returned by any of the editors that support auto-complete, and of the intersection of everything each of them returned. It actually looks like this would be the same as common, except for order. What exactly is the expected behavior when there are multiple auto-completing editors present? Seems like the way it's implemented, it's items that are common to all editors, in the order returned by whichever finished first. If that is the expected behavior, I'd add a comment confirming it, and why that is.

Reply:
> Seems like the way it's implemented, it's items that are common to all editors, in the order returned by whichever finished first.

Correct, I'll add a comment.

Reply:
Comments addressing behavior are on the API